### DISCUSSION TITLE
TMDL GA timeline

### DISCUSSION THREAD

**Discussion by suketududhat** [TabularEditor/TabularEditor3#1370] [🙋♀️TE3 Q&A] [answered]
Hello, the TE3 documentation below indicates that TMDL will be the standard once it is [GA from Microsoft](https://powerbi.microsoft.com/en-us/blog/announcing-general-availability-of-tabular-model-definition-language-tmdl/). Is this announcement a trigger for this standard or are there other dependencies for TE3 that are still in preview? Any timelines on when TMDL would become the standard for TE3? We are assessing whether to transition our models to JSON now or wait for TMDL to be GA in TE3 as well.

![image](https://github.com/user-attachments/assets/7125e88f-feee-412d-8adc-82a7821ebca9)
https://docs.tabulareditor.com/te3/features/tmdl.html#:~:text=To%20enable%20TMDL%20(Preview)%20in,to%20folder...).

**Comment by mlonsk** [comment, marked as answer]
Hi @suketududhat,

I wanted to let you know that we have decided not to deprecate the .json format as we believe it still has its use cases. Thank you for bringing this to our attention. We have updated the documentation accordingly.

Currently, Tabular Editor 3 is using the TOM/AMO library (19.79.1.1), with TMDL in preview. Our plan is to update the TOM/AMO Library to include TMDL GA in our next release, scheduled for October. However, there don't appear to be any changes to TMDL between the Tabular Editor version of the TOM library (19.79.1.1) and the one containing GA of TMDL (19.84.6), so it should be safe to switch already. You can find the source for this information in the following GitHub repo by Mathias Thierbach, who has helped develop TMDL for Microsoft: [mthierba/tmdl-history: Tracks TMDL Preview releases using an example Semantic Model (github.com)](https://github.com/mthierba/tmdl-history?tab=readme-ov-file).

I also recommend keeping a backup of the previous model when converting to TMDL, just in case the conversion fails. Please always ensure that the produced TMDL is valid and can be deployed before continuing development on the TMDL model, as even though something is GA, it does not mean it's bug-free.
